Pipeline still failing in release stage at 'Push commit and tag' task

### Prerequisites

- [x] Write a descriptive title.
- [x] Make sure you are able to reproduce it on the latest released version
- [x] Search the existing issues.

### Steps to reproduce

Same issue, can't push to main

### Expected behavior

```console
Pipeline releases
```

### Actual behavior

```console
Pipeline fails
```

### Error details

```console
Push commit and tag
0s
1
Run git push --follow-tags origin HEAD:main
2
  git push --follow-tags origin HEAD:main
3
  shell: /usr/bin/bash -e {0}
4
  
4
remote: error: GH006: Protected branch update failed for refs/heads/main.        
5
remote: 
6
remote: - Changes must be made through a pull request.        
7
remote: 
8
remote: - 3 of 3 required status checks are expected.        
9
To https://github.com/tonylea/PSSAToJunit
10
 * [new tag]         v0.0.16 -> v0.0.16
11
 ! [remote rejected] HEAD -> main (protected branch hook declined)
12
error: failed to push some refs to 'https://github.com/tonylea/PSSAToJunit'
13
Error: Process completed with exit code 1.
```

### Environment data

```powershell
NA
```

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -23,7 +23,7 @@ jobs:
         with:
           fetch-depth: 0
           ref: ${{ github.event.workflow_run.head_sha }}
-          token: ${{ secrets.GITHUB_TOKEN }}
+          token: ${{ secrets.GH_PAT }}
 
       - name: Configure Git
         run: |

